### ISSUE TITLE
Ceedling does not handle dashes in the module- or filename

### ISSUE THREAD

**Issue by srybalko** [ThrowTheSwitch/Ceedling#439]
### **Problematic:**
Impossible to use ceedling on existing project if filenames have dashes

### **Way to reproduce:**
1. Create project `$ ceedling new dash_test`
2. Go into directory: `$ cd dash_test/`
3. Create module: `$ ceedling module:create[dash-test]`
4. Run test: `$ ceedling`

In the file test/test_dash-test.h:
- Defines are: **DASH-TEST_H**
- Function is: **test_dash-test_NeedToImplement**
In the runner /test_dash-test_runner.c:
run_test(**test_dash-test_NeedToImplement**, "test_dash-test_NeedToImplement", 13);

### **Version:**
```
$ ceedling version
Welcome to Ceedling!
    Ceedling:: 0.29.1
       CMock:: 2.5.1
       Unity:: 2.5.0
  CException:: 1.3.2
```

### Additional info:
[Filename with hyphen
](https://groups.google.com/forum/m/?utm_medium=email&utm_source=footer#!msg/throwtheswitch/nqWOX6fzdGA/f6th-9g-0csJ)
[Ceedling's Unity and CMock not handling spaces or dashes in filenames](https://github.com/ThrowTheSwitch/Ceedling/issues/29)

### **Log:**

$ ceedling new dash_test
```
Welcome to Ceedling!
      create  dash_test/project.yml

Project 'dash_test' created!
 - Execute 'ceedling help' to view available test & build tasks

```
$ cd dash_test/
$ ceedling module:create[dash-test]
```
File src/dash-test.c created
File src/dash-test.h created
File test/test_dash-test.c created
Generate Complete
```
$ ceedling
```
Test 'test_dash-test.c'
-----------------------
In file included from test/test_dash-test.c:3:0:
src/dash-test.h:1:13: warning: extra tokens at end of #ifndef directive
 #ifndef DASH-TEST_H
             ^
src/dash-test.h:2:13: warning: ISO C99 requires whitespace after the macro name
 #define DASH-TEST_H
             ^
Generating runner for test_dash-test.c...
Compiling test_dash-test_runner.c...
build/test/runners/test_dash-test_runner.c:13:22: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘-’ token
 extern void test_dash-test_NeedToImplement(void);
                      ^
build/test/runners/test_dash-test_runner.c: In function ‘main’:
build/test/runners/test_dash-test_runner.c:80:12: error: ‘test_dash’ undeclared (first use in this function)
   run_test(test_dash-test_NeedToImplement, "test_dash-test_NeedToImplement", 13);
            ^~~~~~~~~
build/test/runners/test_dash-test_runner.c:80:12: note: each undeclared identifier is reported only once for each function it appears in
build/test/runners/test_dash-test_runner.c:80:22: error: ‘test_NeedToImplement’ undeclared (first use in this function)
   run_test(test_dash-test_NeedToImplement, "test_dash-test_NeedToImplement", 13);
                      ^~~~~~~~~~~~~~~~~~~~
ERROR: Shell command failed.
> Shell executed command:
'gcc -I"test" -I"test/support" -I"src" -I"/var/lib/gems/2.5.0/gems/ceedling-0.29.1/vendor/unity/src" -I"/var/lib/gems/2.5.0/gems/ceedling-0.29.1/vendor/cmock/src" -I"build/test/mocks" -DTEST -DGNU_COMPILER -g -c "build/test/runners/test_dash-test_runner.c" -o "build/test/out/c/test_dash-test_runner.o" -MMD -MF "build/test/dependencies/test_dash-test_runner.d"'
> And exited with status: [1].

#<Thread:0x0000559e20ba2e30@/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/par_map.rb:7 run> terminated with exception (report_on_exception is true):
Traceback (most recent call last):
	11: from /var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/par_map.rb:10:in `block (2 levels) in par_map'
	10: from /var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/task_invoker.rb:91:in `block in invoke_test_objects'
	 9: from /usr/lib/ruby/vendor_ruby/rake/task.rb:182:in `invoke'
	 8: from /usr/lib/ruby/vendor_ruby/rake/task.rb:193:in `invoke_with_call_chain'
	 7: from /usr/lib/ruby/2.5.0/monitor.rb:226:in `mon_synchronize'
	 6: from /usr/lib/ruby/vendor_ruby/rake/task.rb:213:in `block in invoke_with_call_chain'
	 5: from /usr/lib/ruby/vendor_ruby/rake/task.rb:271:in `execute'
	 4: from /usr/lib/ruby/vendor_ruby/rake/task.rb:271:in `each'
	 3: from /usr/lib/ruby/vendor_ruby/rake/task.rb:271:in `block in execute'
	 2: from /var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/rules_tests.rake:17:in `block in <top (required)>'
	 1: from /var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/generator.rb:99:in `generate_object_file'
/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/tool_executor.rb:88:in `exec': ShellExecutionException (ShellExecutionException)
rake aborted!
ShellExecutionException: ShellExecutionException
/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/tool_executor.rb:88:in `exec'
/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/generator.rb:99:in `generate_object_file'
/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/rules_tests.rake:17:in `block in <top (required)>'
/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/task_invoker.rb:91:in `block in invoke_test_objects'
/var/lib/gems/2.5.0/gems/ceedling-0.29.1/lib/ceedling/par_map.rb:10:in `block (2 levels) in par_map'
Tasks: TOP => build/test/out/c/test_dash-test_runner.o
(See full trace by running task with --trace)
ERROR: Ceedling Failed

```

**Comment by dpostorivo** [comment]
This looks like generator only issues, not necessarily core ceedling issues as the generator creates invalid C functions and defines. If you fix the define name and test name, then the rest of ceedling should work. 

I'll work on a patch for the generator to not to create invalid code in this case soon.

**Comment by srybalko** [comment]
Thanks a lot! I'll change functions and defines names manually for a while

**Comment by mkarlesky** [comment]
This is (finally) fully fixed in 1.0.0
